sidecar/proxy/envoy/manager.go

```diff
@@ -330,65 +333,67 @@ func BuildClusters(instances []*api.ServiceInstance, rules []api.Rule, tlsConfig
 	rules = AddDefaultRouteRules(rules, instances)
 
 	backends := make(map[string]*api.Backend)
+	staticClusters := make(map[string]*api.Backend)
```

Comment:
The logic from line 336 to 409 could be simplified by making `staticClusters` a set.